src/cli/commands/init.js

```diff
@@ -94,7 +94,7 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
     {
       key: 'private',
       question: 'private',
-      default: '',
+      default: String(config.getOption('init-private') || ''),
```

Comment:
This's a `string` value, and (L155)[https://github.com/yarnpkg/yarn/pull/4377/files#diff-527026e7decfe80ca4f832022aeb0d86R155] will pass `string` value of `private` flag to `boolean`

Reply:
Yarn automatically detects "magic" boolean strings `true` and `false` and converts the value into booleans. Try and see yourself ;)